db_migrator--1.1.0.sql

```diff
@@ -43,198 +43,211 @@ BEGIN
 
    EXECUTE format('SET LOCAL search_path = %I, %s', pgstage_schema, extschema);
 
-   /* rename the foreign table */
-   ft := substr(table_name, 1, 62) || E'\x07';
-   stmt := format('ALTER FOREIGN TABLE %I.%I RENAME TO %I', schema, table_name, ft);
-   IF NOT execute_statement(
-      operation => 'rename foreign table',
-      schema => schema,
-      object_name => table_name,
-      stmt => stmt,
-      pgstage_schema => pgstage_schema
-   ) THEN
-      RETURN false;
-   END IF;
+   /* use an explicit transaction block to roll back if any of the steps fail*/
+   BEGIN
+      /* rename the foreign table */
+      ft := substr(table_name, 1, 62) || E'\x07';
+      stmt := format('ALTER FOREIGN TABLE %I.%I RENAME TO %I', schema, table_name, ft);
+      IF NOT execute_statement(
+         operation => 'rename foreign table',
+         schema => schema,
+         object_name => table_name,
+         stmt => stmt,
+         pgstage_schema => pgstage_schema
+      ) THEN
+         ROLLBACK;
```

Comment:
Come on. You cannot have `ROLLBACK` in a function.
You didn't test that, right?

Besides, if you roll back, you'd also roll back the log entry.

The options I see is to either not refactor that function or to explicitly undo any modifications (drop table, rename back) if you hit an error condition.

Reply:
I didn't test anything with `ora_migrator` until now. I just managed to install an Oracle instance to reproduce the error. I'll experiment another way to call `execute_statement()`. Thank you for catching all theses transactions issues.